config/core/src/test/groovy/org/apache/shiro/config/IniTest.groovy

```diff
@@ -119,24 +135,40 @@ public class IniTest {
         assertEquals("Truth", kv[0]);
         assertEquals("Beauty", kv[1]);
 
+        // Escape characters are to be removed from the key.
+        // This is different behaviour compared to the XML config.
         test = "Tru\\th=Beauty";
         kv = Ini.Section.splitKeyValue(test);
         assertEquals("Truth", kv[0]);
         assertEquals("Beauty", kv[1]);
 
-        test = "Truth\\=Beauty";
+        // SHIRO-530: Keep backslashes in value.
+        test = "Truth=Beau\\ty";
         kv = Ini.Section.splitKeyValue(test);
         assertEquals("Truth", kv[0]);
-        assertEquals("Beauty", kv[1]);
+        assertEquals("Beau\\ty", kv[1]);
 
-        test = "Truth=Beau\\ty";
+        // SHIRO-530: Keep backslashes in value.
+        test = "Truth=Beauty\\";
```

Comment:
shouldn't this one wrap? and the next line would be included?
i.e. something like:
```ini
Truth=Beauty\
more text here
```
Which would be something like `Truth=Beatuty\\\nmore text here`

Reply:
No, it should not, because we are calling the `splitKeyValue` method. The backslashes at the end are resolved in the caller method. When arriving at this point, the wrapping would have already occured. Took me a while, too, to figure this one out. But I did not bother to add more comments. ;-)

Reply:
@bdemers see here: https://github.com/apache/shiro/blob/b50b829a285b106666b688a4b69a3ebee94f51b4/config/core/src/main/java/org/apache/shiro/config/Ini.java#L609-L631

Please check if I did not mis-read it.